ls1intum/Artemis
# [Refactor] small improvements for grading instruction details component

### Checklist

### Motivation and Context



This PR aims refactoring for grading-instruction-details-component. Currently, reset button for grading criteria title and grading instruction does not work if it is newly added by user. Because, I was comparing the backup exercise and the current exercise, so if the object is not part of backup exercise, reset did not work. To fix, I just initialize with default title and grading instruction if user wants to reset changes for newly added grading criteria or grading instruction. In addition, I added tooltip for reset/delete/plus button in the form.
I also changed some strings to keep consistency.

### Description


### Steps for Testing

P.S. _To display the grading instruction in this form, you need to use the grading instruction as a feedback during the assessment at least once, probably we will provide this mode even if the grading instruction is not associated with the feedback. However, its not scope of this PR._ 

1. Log in to Artemis
2. Navigate to Course Administration
3. Create/Find exercise.
4. Go to edit view. Switch Grading Instructions.
5. Hover the buttons for grading instructions, you should see the tooltip. 
6. Add new grading criterion/grading instruction, change something then click the reset button, it should display default text.
7. Change title for newly added grading criteria then click small reset button, it should display only 'Criterion Title' placeholder for input area.

### Test Coverage



nothing changed.

### Screenshots



<img src="https://user-images.githubusercontent.com/23650630/123876736-2069c580-d93c-11eb-935d-5b9e718a94b5.png" width="500" />
